File: QTD.SPA/src/app/_DtoModels/Version_Tool/Version_Tool.ts
import { Entity } from '../Entity';
import { Tool } from '../Tool/Tool';
import { Version_EnablingObjective_Tool_Link } from '../Version_EnablingObjective_Tool_Link/Version_EnablingObjective_Tool_Link';
import { Version_Procedure_Tool_Link } from '../Version_Procedure_Tool_Link/Version_Procedure_Tool_Link';
import { Version_Task_Tool_Link } from '../Version_Task_Tool_Link/Version_Task_Tool_Link';

export class Version_Tool extends Entity {
  ToolId!: any;
  Description!: string;
  MinorVersion!: number;
  MajorVersion!: number;
  Tool!: Tool;
  Version_Procedure_Tool_Links!: Version_Procedure_Tool_Link[];
  Version_Task_Tool_Links!: Version_Task_Tool_Link[];
  Version_EnablingObjective_Tool_Links!: Version_EnablingObjective_Tool_Link[];
}

File: QTD2.Domain/Entities/Core/Version_Tool.cs
using System.Collections.Generic;

namespace QTD2.Domain.Entities.Core
{
    public class Version_Tool : Common.Entity
    {
        public int ToolId { get; set; }

        public string Description { get; set; }

        public int MinorVersion { get; set; }

        public int MajorVersion { get; set; }

        public virtual Tool Tool { get; set; }

        public virtual ICollection<Version_Procedure_Tool_Link> Version_Procedure_Tool_Links { get; set; } = new List<Version_Procedure_Tool_Link>();

        public virtual ICollection<Version_Task_Tool_Link> Version_Task_Tool_Links { get; set; } = new List<Version_Task_Tool_Link>();

        public virtual ICollection<Version_EnablingObjective_Tool_Link> Version_EnablingObjective_Tool_Links { get; set; } = new List<Version_EnablingObjective_Tool_Link>();

        public Version_Tool()
        {
        }

        public Version_Tool(Tool tool)
        {
            ToolId = tool.Id;
            Description = tool.Name;
        }
    }
}
